src/app/core/duplicate-records/duplicate-records.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service'; 
+import { EntityRegistry } from '../entity/database-entity.decorator';
+import { EntitySchemaService } from '../entity/schema/entity-schema.service';
+import { Entity } from '../entity/model/entity';
+import { MatSnackBar, MatSnackBarHorizontalPosition, MatSnackBarVerticalPosition } from '@angular/material/snack-bar';
+import { ConfigService } from '../config/config.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class DuplicateRecordService {
+  get: jasmine.Spy<jasmine.Func>;
+  constructor(
+    private entitymapperservice: EntityMapperService,
+    private entityTypes: EntityRegistry,
+    private entityService: EntitySchemaService,
+    private snackBar: MatSnackBar,
+    private config: ConfigService
+  ) {}
+
+  showFlashMessage(message: string) {
+    const horizontalPosition: MatSnackBarHorizontalPosition = 'center';
+    const verticalPosition: MatSnackBarVerticalPosition = 'top';
+    this.snackBar.open(message, 'Close', {
+      duration: 3000, 
+      horizontalPosition: horizontalPosition,
+      verticalPosition: verticalPosition,
+    });
+  }
+
+  async duplicateRecord(sourceData: any, schemaName: string) {
```

Comment:
can you find the exact type instead of using `any`?

Reply:
@sleidig, I have changed with `Entity[]`